query-request.hh

```diff
@@ -205,6 +205,9 @@ public:
     const std::unique_ptr<specific_ranges>& get_specific_ranges() const {
         return _specific_ranges;
     }
+    std::unique_ptr<specific_ranges>& get_specific_ranges() {
+        return _specific_ranges;
+    }
```

Comment:
These mutable accessors are tricky. It's easy to call foo(my_obj.get_specific_ranges()) and not realize it's mutable (esp. since the name says "get"). It's also hard to enforce invariants, since the class doesn't have an opportunity to check/adjust its internals after the property has changed. It's often better to have a setter instead.

It's fine to keep as is, it's not a read flag.

Reply:
I'll check if I can replace these with appropriate methods in `slice_builder`. Maybe accessors are all const for a reason, to make the slice effectively immutable (although the public `_ranges` member kindof subverts this).